sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -667,10 +715,12 @@ public void validate(PBegin input) {
 
       @Override
       public PCollection<TableRow> expand(PBegin input) {
-        String uuid = randomUUIDString();
-        final String jobIdToken = "beam_job_" + uuid;
-
+        stepUuid = randomUUIDString();
         BigQueryOptions bqOptions = input.getPipeline().getOptions().as(BigQueryOptions.class);
+        jobUuid = NestedValueProvider.of(
+           StaticValueProvider.of(bqOptions.getJobName()), new CreatePerBeamJobUuid(stepUuid));
```

Comment:
Right, this is step one, and VP-ifying job name (slash, maybe job ID) would be step two.